pkg/apis/kops/validation/legacy.go

```diff
@@ -76,14 +76,24 @@ func ValidateCluster(c *kops.Cluster, strict bool) *field.Error {
 
 	requiresSubnets := true
 	requiresNetworkCIDR := true
+	requiresSubnetCIDR := true
 	switch kops.CloudProviderID(c.Spec.CloudProvider) {
 	case kops.CloudProviderBareMetal:
 		requiresSubnets = false
 		requiresNetworkCIDR = false
+		if c.Spec.NetworkCIDR != "" {
+			return field.Invalid(fieldSpec.Child("NetworkCIDR"), c.Spec.NetworkCIDR, "NetworkCIDR should not be set on bare metal")
+		}
+
+	case kops.CloudProviderGCE:
```

Comment:
Same comment as above.  Can we break this into a different PR?

Reply:
Will move the bare metal one into another PR, but it's not really practical to split out the GCE logic here..

Reply:
Done in #3495